src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -1170,48 +1174,56 @@ public static void Reverse(ref byte buf, nuint length)
                     tempLast = Avx2.Permute2x128(tempLast, tempLast, 0b00_01);
 
                     // Store the reversed vectors
-                    tempLast.StoreUnsafe(ref buf, firstOffset);
-                    tempFirst.StoreUnsafe(ref buf, lastOffset);
-                }
-                buf = ref Unsafe.Add(ref buf, numIters * numElements);
-                length -= numIters * numElements * 2;
+                    tempLast.StoreUnsafe(ref first);
+                    tempFirst.StoreUnsafe(ref last);
+
+                    first = ref Unsafe.Add(ref first, Vector256<byte>.Count);
+                    last = ref Unsafe.Subtract(ref last, Vector256<byte>.Count);
+                } while (Unsafe.IsAddressLessThan(ref first, ref last));
+
+                remainder = Unsafe.ByteOffset(ref first, ref Unsafe.Add(ref last, Vector256<byte>.Count)) / sizeof(byte);
             }
-            else if (Vector128.IsHardwareAccelerated && (nuint)Vector128<byte>.Count * 2 <= length)
+            else if (remainder >= sizeof(long))
```

Comment:
Sounds good, I'll take a look again on Friday

Reply:
In your example the 63 range is indeed covered by the AVX2 loop in one iteration, but since they overlapped the remainder to ends up being -1 :)
However, for example the range between 68-96 would fall-through, so it would still be worthwhile